File: CSharp-Part-1/5. Conditional Statements/ConditionalStatements/DigitAsWord/DigitAsWord.cs
﻿/*
 * Write a program that asks for a digit (0-9), and depending on the input, shows the digit as a word (in English).
 * Print “not a digit” in case of invalid input.
 * Use a switch statement.
 */

using System;

class DigitAsWord
{
    static void Main()
    {
        Console.WriteLine("Enter digit: ");
        string inputDigit = Console.ReadLine();

        int digit;

        bool result = Int32.TryParse(inputDigit, out digit);
        if (result)
        {
            switch (digit)
            {
                case 0:
                    Console.WriteLine("Zero.");
                    break;
                case 1:
                    Console.WriteLine("One.");
                    break;
                case 2:
                    Console.WriteLine("Two.");
                    break;
                case 3:
                    Console.WriteLine("Three.");
                    break;
                case 4:
                    Console.WriteLine("Four.");
                    break;
                case 5:
                    Console.WriteLine("Five.");
                    break;
                case 6:
                    Console.WriteLine("Six.");
                    break;
                case 7:
                    Console.WriteLine("Seven.");
                    break;
                case 8:
                    Console.WriteLine("Eight.");
                    break;
                case 9:
                    Console.WriteLine("Nine.");
                    break;
                default:
                    Console.WriteLine("Not a digit.");
                    break;
            }
        }
        else
        {
            Console.WriteLine("Not a digit.");
        }
    }
}

File: JavaScript-Fundamentals/05. Conditional Statements/digit-as-word.js
/*
	Write a script that asks for a digit (0-9), and depending on the input, shows the digit as a word (in English).
	Print “not a digit” in case of invalid input.
	Use a switch statement.
*/

function wordify(digit) {
	switch (digit) {
		case 0:
			return 'zero';
		case 1:
			return 'one';
		case 2:
			return 'two';
		case 3:
			return 'three';
		case 4:
			return 'four';
		case 5:
			return 'five';
		case 6:
			return 'six';
		case 7:
			return 'seven';
		case 8:
			return 'eight';
		case 9:
			return 'nine';
		default:
			return 'not a digit';
	}
}

for (var digit = 0; digit <= 10; digit += 1) {
	console.log(wordify(digit));
}

console.log(wordify(-0.1));
console.log(wordify('hi'));